src/server/pps/server/api_server.go

```diff
@@ -2135,6 +2135,29 @@ func (a *apiServer) CreatePipelineV2(ctx context.Context, request *pps.CreatePip
 	}, nil
 }
 
+func (a *apiServer) RerunPipeline(ctx context.Context, request *pps.RerunPipelineRequest) (response *emptypb.Empty, err error) {
```

Comment:
Looking at the implementation of the helpers here, I don't think we need to do this much work.  I think start a write transaction, inspect the pipeline, change salt to a new uuid4withoutdashes, write the pipeline back with updatePipeline.

Reply:
But in general, don't do the read -> edit -> write outside of a transaction.  It can destroy data that is changed while the edit is ongoing (perhaps in another process).

Reply:
Paired with Jonathan here on [this commit](https://github.com/pachyderm/pachyderm/pull/9287/commits/1f5d8918d9a516153412e7a69406d1d9dccf6dbb).